features/nvstore/source/nvstore_int_flash_wrapper.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+// ----------------------------------------------------------- Includes -----------------------------------------------------------
+
+#ifdef DEVICE_FLASH
+
+#include "nvstore_int_flash_wrapper.h"
+#include "nvstore_shared_lock.h"
+#include "FlashIAP.h"
+#include <stdlib.h>
+
+// --------------------------------------------------------- Definitions ----------------------------------------------------------
+
+#define MIN(a,b)            ((a) < (b) ? (a) : (b))
+#define MAX(a,b)            ((a) > (b) ? (a) : (b))
+
+#define MAX_PAGE_SIZE   16
+
+static mbed::FlashIAP flash;
```

Comment:
Can the linker remove this if NVStore is not used?

Reply:
After folding this file into NVStore.cpp (see my previous answer about it), this will be done by default (as NVStore code is only built when the feature is enabled). In addition, will allocate an instance of it dynamically on init.
On a side note, I think FlashIAP should be implemented as a singleton class. Don't think that having multiple instances of this class makes much sense.